Makefile

```
@@ -0,0 +1,39 @@
+# several pytest settings
+DURATION ?= all  # pytest duration, one of short, long or all
+WORLD_SIZE ?= 1  # world size for DDP tests
+EXTRA_ARGS ?=  # additional arguments
+
+EXTRA_ARGS := --duration $(DURATION) $(EXTRA_ARGS)
+
+dirs := composer examples tests
+
+# run this to autoformat your code
+style:
+	isort -i $(dirs)
+	yapf -ri $(dirs)
+	docformatter -ri --wrap-summaries 120 --wrap-descriptions 120 $(dirs)
+
+# this only checks for style & pyright, makes no code changes
+lint:
+	isort $(dirs) -c
+	yapf -dr $(dirs)
+	docformatter -r --wrap-summaries 120 --wrap-descriptions 120 $(dirs)
+	pyright $(dirs)
+
+test:
+	pytest tests/ $(EXTRA_ARGS)
+
+test-gpu:
+	pytest tests/ -m gpu $(EXTRA_ARGS)
+
+test-deepspeed:
+	pytest tests/ -m deepspeed $(EXTRA_ARGS)
+
+# run all tests, including multi-gpu tests
+# uses the composer launcher script
+test-ddp:
```

nit: `ddp` refers to a specific distributed backend, so we started to call it `dist` instead

```suggestion
test-dist:
```

---

Unfortunately, `dist` is usually the shorthand for `torch.distributed`, which is synonymous with `ddp` so not sure the right name here.

---

Both DDP and Deepspeed are distributed strategies, and both use the launcher script, so I think `dist` would work here?